lms/djangoapps/teams/csv.py

```diff
@@ -292,18 +300,21 @@ def validate_user_assignment_to_team_and_teamset(self, row):
                     return False
                 else:
                     self.user_ids_by_teamset_id[teamset_id].add(user.id)
-                    continue
-            max_team_size = self.course.teams_configuration.default_max_team_size
-            if max_team_size is not None and team.users.count() >= max_team_size:
-                if self.add_error_and_check_if_max_exceeded('Team ' + team.team_id + ' is full.'):
-                    return False
+            if not self.validate_proposed_team_size_wont_exceed_maximum(team_name, teamset_id, current_team_size):
+                return False
+        return True
 
-            if (user.id, team.topic_id) in self.existing_course_team_memberships:
-                error_message = 'User {0} is already on a team in teamset {1}.'.format(
-                    user.username, team.topic_id
-                )
-                if self.add_error_and_check_if_max_exceeded(error_message):
-                    return False
+    def validate_proposed_team_size_wont_exceed_maximum(self, team_name, teamset_id, current_team_size):
+        """
+        Validates that the number of users we want to add to a team won't exceed maximum team size.
+        """
+        max_team_size = self.course.teams_configuration.default_max_team_size
```

Comment:
Also, don't we want the `max_team_size` for the given `teamset_id`.  I think `default_max_team_size` is the default for when a teamset doesn't have an explicitly configured max.